hail/src/main/scala/is/hail/expr/ir/Simplify.scala

```diff
@@ -193,14 +193,16 @@ object Simplify {
 
     case ArrayLen(MakeArray(args, _)) => I32(args.length)
 
-    case ArrayLen(ToArray(StreamMap(s, _, _))) => ArrayLen(ToArray(s))
+    case ArrayLen(ToArray(StreamMap(s, _, _))) => StreamLen(s)
+    case StreamLen(StreamMap(s, _, _)) => StreamLen(s)
```

Comment:
add one that rewrites streamlen(streamflatmap to a Streamfold that takes sum of streammap(x => streamlen(x)?

Reply:
that may optimize better

Reply:
Sure, good idea

Reply:
Yeah that's a good one. Not just optimize better, better codegen.

Reply:
Done